### ISSUE TITLE
OSX can't break up words in Arabic

### ISSUE THREAD

**Issue by alerque** [simoncozens/sile#123]
So something is wrong on OSX. I've been working through the test suite and while many of the regression tests work fine, none of the ones involving Arabic work right. More specifically any of them involving long lines don't know when to wrap because apparently words aren't getting split right.

![ekran resmi 2015-06-30 21 12 44](https://cloud.githubusercontent.com/assets/173595/8438207/1df6af64-1f6d-11e5-9e8c-a134cc3553be.png)

A quick look at the debug output shows the expected output is a bunch of items broken by word, but the actual output is broken up more by sentence. Here is a bit of diff:

![ekran resmi 2015-06-30 21 17 07](https://cloud.githubusercontent.com/assets/173595/8438286/80d128bc-1f6d-11e5-80ae-580554ddf426.png)

**Comment by simoncozens** [comment]
I develop on OSX, and that works fine. I suspect (again) Lua 5,3 and the Unicode library.

**Comment by simoncozens** [comment]
Could you just try replacing "pcall(function() utf8 = require("utf8") end)" with "utf8 = nil" or similar and see if our native functions split things up better?

**Comment by alerque** [comment]
he result shown in this issue was encountered on my OSX test rig which has a stock setup including homebrew's default lua—5.2.

* Up until a minute ago it did not have the utf8 luarock installed. Now it
  does, and I tested again. The resulting PDF & debug output are
  identical. Apparently it wasn't used?

* The change you suggested (patch below to confirm I did it right) also
  had zero effect.

		diff --git a/core/utilities.lua b/core/utilities.lua
		index d35182e..744f694 100644
		--- a/core/utilities.lua
		+++ b/core/utilities.lua
		@@ -125,7 +125,8 @@ function utilities.allCombinations(options)
		end
		
		-- Unicode-related utilities
		-pcall(function() utf8 = require("utf8") end)
		+--pcall(function() utf8 = require("utf8") end)
		+utf8 = nil
		if not utf8 then utf8 = {} end
		
		utilities.utf8char = utf8.char or function (c)

My conclusion is that it's whatever you referred to as "our native
functions" that are already in play for me and not doing their job well.
And for some reason even with the luarock installed I'm getting it to pick
up on whatever library it's using on your end.

**Comment by simoncozens** [comment]
OK, so it's not luautf8 versus utf8 versus our Unicode functions in `utilities.lua`.

I wonder if it's related to the new Unicode tokenizer. Could you try this at a `./sile` REPL:

```
for i in SILE.tokenizers.unicode("عندما يريد") do print(i) end
```

I get:

```
{string=عندما}
{separator= }
{string=يريد}
```

**Comment by alerque** [comment]
On my own Linux machine I see what you see. On the OSX machine I'm testing with (Lua 5.2) I see:

    {string=عندما يريد}

**Comment by simoncozens** [comment]
OK. Here's a daft thing: did you install SILE on OSX before I added the Unicode tokenizer, and is it finding an old version of `char-def.lua`? (The package path goes `...;/usr/local/share/sile/lua-libraries/?/init.lua;lua-libraries/?.lua;...` so it will find an installed version before a local version.)

**Comment by alerque** [comment]
Huh. So ya. I have never installed SILE on this machine except through Homebrew, but I did have both the stable and the HEAD versions installed and available. That left me with a sum total of these copies of that lib:

	/Library/Caches/Homebrew/sile--git/lua-libraries/char-def.lua
	/Users/caleb/projects/sile/lua-libraries/char-def.lua
	/usr/local/Cellar/sile/0.9.2/share/sile/lua-libraries/char-def.lua
	/usr/local/Cellar/sile/HEAD/share/sile/lua-libraries/char-def.lua
	/usr/local/texlive/2013/texmf-dist/tex/context/base/char-def.lua
	/usr/local/texlive/2014/texmf-dist/tex/context/base/char-def.lua
	/usr/local/texlive/2014/texmf-dist/tex/lualatex/simurgh/simurgh-char-def.lua

Go figure, even though my path is such that the HEAD installation was the first in line, whether I ran that installation or locally as `./sile` in my project directory it was finding the char-def library from the 0.9.2 installation. Uninstalling that copy makes it work and I get properly tokenized Arabic strings.

Is there something I should do with the Homebrew installations so they prefer the library that's closest to them?

Now I'm just left with the fact that there seem to be different versions of Scheherazade out there with different metrics (which is why #122 is failing to pass tests right now).
